src/rand.rs

```diff
@@ -80,10 +89,29 @@ fn rand_alphanumerics_impl(rand_fn: &dyn Fn() -> u32, len: usize) -> OsString {
     }
 }
 
+fn rand_bytes_impl(rand_fn: &dyn Fn() -> u32, len: usize) -> Vec<u8> {
+    let mut buf: Vec<u8> = Vec::new();
```

Comment:
The code in this module looks a bit C-like, and should probably be refactored in a future iteration, but there is no reason to continue the trend for new functions as well. Can you try to use more idiomatic ways of initializing the vector? For example, you can combine `iter` with `take` to retrieve the number of elements you need from a vector. Alternatively, you check how many times you have to call the `xor_pseudo_rng..` function and use a combination of `iter`, `map`, and `zip`.